pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -263,8 +222,11 @@ func (a *Adapter) Export(ctx context.Context) (*unstructured.Unstructured, error
 	if mapCtx.Err() != nil {
 		return nil, mapCtx.Err()
 	}
-	obj.Spec.ProjectRef = &refs.ProjectRef{Name: a.id.Parent.Project}
-	obj.Spec.Location = a.id.Parent.Location
+	tokens := strings.Split(a.id.External, "/")
+	if len(tokens) == 6 {
+		obj.Spec.ProjectRef = &refs.ProjectRef{Name: tokens[1]}
```

Comment:
We could define some helper methods on the ref also, e.g. `ProjectID` and `Location`.  For 90%+ of our Refs (I would guess), they can delegate to a method like `extractKey("projects")` or `extractKey("locations")`